h2/src/main/org/h2/engine/Database.java

```diff
@@ -690,7 +678,7 @@ private void executeMeta(ArrayList<MetaRecord> records) {
     }
 
     private void handleUpgradeIssues() {
-        if (store != null && !isReadOnly()) {
+        if (!isReadOnly()) {
```

Comment:
Perhaps the whole this method is now useless, because H2 refuses to open older database files.

Reply:
Sure